ckan/config/config_declaration.yaml

```diff
@@ -258,99 +242,141 @@ groups:
         description: |
-          What domain the cookie should be set to. When using sub-domains,
-          this should be set to the main domain the cookie should be valid for. For example,
-          if a cookie should be valid under www.nowhere.com and files.nowhere.com then it should be set to .nowhere.com.
-          Defaults to the current domain in its entirety.
+          The path that the session cookie will be valid for.
 
-      - key: beaker.session.save_accessed_time
+      - key: SESSION_COOKIE_HTTPONLY
         type: bool
-        default: True
-        description: Whether beaker should save the session's access time (true) or only modification time (false).
+        default: true
+        description: |
+          Browsers will not allow JavaScript access to cookies marked as “HTTP only” for security.
 
-      - key: beaker.session.secure
+      - key: SESSION_COOKIE_SECURE
         type: bool
         description: |
-          Whether or not the session cookie should be marked as secure. When marked as secure,
-          browsers are instructed to not send the cookie over anything other than an SSL connection.
+          Browsers will only send cookies with requests over HTTPS if the
+          cookie is marked “secure”. The application must be served over HTTPS
+          for this to make sense.
 
-      - key: beaker.session.timeout
-        validators: int_validator
+      - key: SESSION_COOKIE_SAMESITE
+        default: Lax
+        validators: one_of(["Strict","Lax","None"])
         description: |
-          Seconds until the session is considered invalid, after which it will be ignored and invalidated.
-          This number is based on the time since the session was last accessed, not from when the session was created.
-          Defaults to never expiring.
-          Requires that save_accessed_time be true.
+          Restrict how cookies are sent with requests from external sites. Can
+          be set to 'Lax' (recommended) or 'Strict'
 
-      - key: beaker.session.type
-        default: cookie
+      - key: SESSION_REFRESH_EACH_REQUEST
+        type: bool
         description: |
-          The type of session to use. The default is cookie, which uses a cookie to store the session id.
-          Other options include file, which stores the session id in a file, and dbm, which stores the session id in a dbm file.
-          The dbm option is not recommended as it is not thread-safe.
+          Control whether the cookie is sent with every response when
+          SESSION_PERMANENT is true. Sending the cookie every time can more
+          reliably keep the session from expiring, but uses more
+          bandwidth. Non-permanent sessions are not affected.
 
-      - key: beaker.session.validate_key
-        validators: not_empty
-        placeholder_callable: secrets:token_urlsafe
-        validators: configured_default("SECRET_KEY",None)
-        callable_args:
-          nbytes: 20
+      - key: SESSION_PERMANENT
```

Comment:
Yes, I am also interested in why you are so sure, that ours defaults to the browser's session?:)

Reply:
The config description says ` When disabled, session expires with the browser session.` That's why I am asking why is that behavior instead of Flask's default.

Reply:
PS and you mention that "our defaults to browser session". I suspect that you were just distracted by someone from a review of this PR, so let us see why it's not so:

```yaml
- key: SESSION_PERMANENT
  default: true
```
Looks like by default this option IS enabled. Let's check what the description says:

>  When **disabled**, the session expires with the browser session. When **enabled**, the session's expiration time is set to ``PERMANENT_SESSION_LIFETIME`` seconds.

The first sentence that mentions browser session is not our case, because we enable this flag by default. The second sentence explains that our session's lifetime is controlled by `PERMANENT_SESSION_LIFETIME`. So, yes, we are not controlled by browser sessions by default.